coderd/provisionerdserver/provisionerdserver_test.go

```diff
@@ -1804,6 +1804,168 @@ func TestNotifications(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("TemplateManualBuildFailed", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("NotifyTemplateOwner", func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+			notifEnq := &testutil.FakeNotificationsEnqueuer{}
+
+			//	Otherwise `(*Server).FailJob` fails with:
+			// audit log - get build {"error": "sql: no rows in result set"}
+			ignoreLogErrors := true
+			srv, db, ps, pd := setup(t, ignoreLogErrors, &overrides{
+				notificationEnqueuer: notifEnq,
+			})
+
+			initiator := dbgen.User(t, db, database.User{})
+			templateOwner := dbgen.User(t, db, database.User{})
+
+			template := dbgen.Template(t, db, database.Template{
+				Name:           "template",
+				Provisioner:    database.ProvisionerTypeEcho,
+				OrganizationID: pd.OrganizationID,
+				CreatedBy:      templateOwner.ID,
+			})
+			template, err := db.GetTemplateByID(ctx, template.ID)
+			require.NoError(t, err)
+			file := dbgen.File(t, db, database.File{CreatedBy: initiator.ID})
+			workspace := dbgen.Workspace(t, db, database.Workspace{
+				TemplateID:     template.ID,
+				OwnerID:        initiator.ID,
+				OrganizationID: pd.OrganizationID,
+			})
+			version := dbgen.TemplateVersion(t, db, database.TemplateVersion{
+				OrganizationID: pd.OrganizationID,
+				TemplateID: uuid.NullUUID{
+					UUID:  template.ID,
+					Valid: true,
+				},
+				JobID: uuid.New(),
+			})
+			build := dbgen.WorkspaceBuild(t, db, database.WorkspaceBuild{
+				WorkspaceID:       workspace.ID,
+				TemplateVersionID: version.ID,
+				InitiatorID:       initiator.ID,
+				Transition:        database.WorkspaceTransitionDelete,
+				Reason:            database.BuildReasonInitiator,
+			})
+			job := dbgen.ProvisionerJob(t, db, ps, database.ProvisionerJob{
+				FileID: file.ID,
+				Type:   database.ProvisionerJobTypeWorkspaceBuild,
+				Input: must(json.Marshal(provisionerdserver.WorkspaceProvisionJob{
+					WorkspaceBuildID: build.ID,
+				})),
+				OrganizationID: pd.OrganizationID,
+			})
+			_, err = db.AcquireProvisionerJob(ctx, database.AcquireProvisionerJobParams{
+				OrganizationID: pd.OrganizationID,
+				WorkerID: uuid.NullUUID{
+					UUID:  pd.ID,
+					Valid: true,
+				},
+				Types: []database.ProvisionerType{database.ProvisionerTypeEcho},
+			})
+			require.NoError(t, err)
+
+			_, err = srv.FailJob(ctx, &proto.FailedJob{
+				JobId: job.ID.String(),
+				Type: &proto.FailedJob_WorkspaceBuild_{
+					WorkspaceBuild: &proto.FailedJob_WorkspaceBuild{
+						State: []byte{},
+					},
+				},
+			})
+			require.NoError(t, err)
+
+			require.Len(t, notifEnq.Sent, 1)
+			require.Equal(t, notifEnq.Sent[0].UserID, templateOwner.ID)
```

Comment:
Please check for the template name to ensure we're receiving the expected notification.